### PR TITLE
Add cut,copy and paste functionality.

### PR DESCRIPTION
**Pull request by kristijanhusak**
This PR adds cut, copy and paste functionality that I mentioned in issue https://github.com/kyazdani42/nvim-tree.lua/issues/51

Note that I don't have much experience with Lua, so do as detailed review as possible.

### CODE REVIEW

**Comment by kyazdani42** [comment]
thanks :) I'll review this PR in depth later but it looks already pretty good imo.

**Comment by kristijanhusak** [comment]
I think handling of existing files on copy/move needs to be handled. Now it will just throw an error, but I guess it should prompt for override/cancel. Will try to add it.

File: lua/lib/fs.lua [hunk 1]
```diff
@@ -109,6 +113,92 @@ local function remove_dir(cwd)
...
+      api.nvim_err_writeln('Could not '..action_type..' '..entry.absolute_path..' - '..msg)
+    end
+    ::continue::
+  end
+  clipboard[action_type] = {}
+  return refresh_tree()
+end
+
+local function add_to_clipboard(node, clip)
+  if node.name == '..' then return end
```
Comment (kyazdani42) [inline lua/lib/fs.lua:190]: this should be also checked when pasting or it will error out.

File: lua/lib/fs.lua [hunk 2]
```diff
@@ -109,6 +113,92 @@ local function remove_dir(cwd)
...
+      local ans = vim.fn.input(dest..' already exists, overwrite ? y/n: ')
+      clear_prompt()
+      if not ans:match('^y') then goto continue end
+    end
+
+    local success, msg = action_fn(entry.absolute_path, dest)
+    if not success then
+      api.nvim_err_writeln('Could not '..action_type..' '..entry.absolute_path..' - '..msg)
+    end
+    ::continue::
```
Comment (kyazdani42) [inline lua/lib/fs.lua:183]: goto is not available for some distributions of neovim (as neovim uses lua 5.1 where goto is not available). You could leave it for now as i still have some piece of code with `goto`s but this issue will need to be addressed in another PR.

File: lua/lib/fs.lua (hunk 1 above)
Comment (kristijanhusak) [reply lua/lib/fs.lua:193]: Added it to `do_paste` function.

File: lua/lib/fs.lua (hunk 2 above)
Comment (kristijanhusak) [reply lua/lib/fs.lua]: I changed the logic to not use continue.

**Comment by kristijanhusak** [comment]
Yeah I can add something. Maybe echo will be sufficient?
Do you want to have it like a mapping or command?

**Comment by kyazdani42** [comment]
a command would be more appropriate. Echo should be enough indeed

**Comment by kristijanhusak** [comment]
I added `:LuaTreeClipboard` command. See if the format is ok.
